static/app/views/automations/components/automationBuilder.tsx

```diff
@@ -11,25 +11,139 @@ import {IconAdd, IconDelete, IconMail} from 'sentry/icons';
 import {useAutomationBuilderContext} from 'sentry/views/automations/components/automationBuilderContext';
 import DataConditionNodeList from 'sentry/views/automations/components/dataConditionNodeList';
 import {TRIGGER_MATCH_OPTIONS} from 'sentry/views/automations/components/triggers/constants';
 
+const frequencyTypes = [
+  DataConditionType.EVENT_FREQUENCY_COUNT,
+  DataConditionType.EVENT_FREQUENCY_PERCENT,
+  DataConditionType.EVENT_UNIQUE_USER_FREQUENCY_COUNT,
+  DataConditionType.EVENT_UNIQUE_USER_FREQUENCY_PERCENT,
+];
+
+function findConflictingActionFilterConditions(
+  actionFilter: DataConditionGroup
+): string[] {
+  const incompatibleConditions = [];
+
+  // Find incompatible conditions for NONE logic type
+  if (actionFilter.logicType === DataConditionGroupLogicType.NONE) {
+    for (const condition of actionFilter.conditions) {
+      const isInvalidAgeComparison =
+        condition.type === DataConditionType.AGE_COMPARISON &&
+        condition.comparison.comparison_type === AgeComparison.NEWER &&
+        condition.comparison.value > 0;
+      const isInvalidIssueOccurence =
+        condition.type === DataConditionType.ISSUE_OCCURRENCES &&
+        condition.comparison.value <= 1;
+
+      if (isInvalidAgeComparison || isInvalidIssueOccurence) {
+        incompatibleConditions.push(condition.id);
+      }
+      return incompatibleConditions;
+    }
+  }
+
+  // Find incompatible conditions for ANY_SHORT_CIRCUIT and ALL logic types
+  for (const condition of actionFilter.conditions) {
+    const isInvalidFrequency =
+      frequencyTypes.includes(condition.type) && condition.comparison.value >= 1;
+    const isInvalidAgeComparison =
+      condition.type === DataConditionType.AGE_COMPARISON &&
+      condition.comparison.comparison_type === AgeComparison.OLDER;
+    const isInvalidIssueOccurence =
+      condition.type === DataConditionType.ISSUE_OCCURRENCES &&
+      condition.comparison.value > 1;
+
+    if (isInvalidFrequency || isInvalidAgeComparison || isInvalidIssueOccurence) {
+      incompatibleConditions.push(condition.id);
+    }
+  }
+
+  // If the logic type is ANY_SHORT_CIRCUIT and any of the conditions are valid, consider the action filter valid
+  if (
+    actionFilter.logicType === DataConditionGroupLogicType.ANY_SHORT_CIRCUIT &&
+    incompatibleConditions.length !== actionFilter.conditions.length
+  ) {
+    return [];
+  }
+
+  return incompatibleConditions;
+}
+
 export default function AutomationBuilder() {
   const {state, actions} = useAutomationBuilderContext();
   const organization = useOrganization();
   const api = useApi();
 
-  // fetch org members for SelectMembers dropdowns
+  // Fetch org members for SelectMembers dropdowns
   useEffect(() => {
     fetchOrgMembers(api, organization.slug);
   }, [api, organization]);
 
+  const {conflictingTriggers, conflictingActionFilters} = useMemo((): {
+    conflictingActionFilters: Record<string, string[]>;
+    conflictingTriggers: string[];
```

Comment:
might be worth pulling this out so we can unit test this function. I know that helped a lot with development of the previous version

Reply:
yeah i can probably pull more of it out, so we can add tests for trigger + action filter combinations